packages/dev/core/src/Rendering/IBLShadows/iblShadowsAccumulationPass.ts

```diff
@@ -0,0 +1,254 @@
+        };
+        this._oldAccumulationRT.addPostProcess(accumulationCopyPP);
+        this._oldAccumulationRT.skipInitialClear = true;
+        this._oldAccumulationRT.noPrePassRenderer = true;
+        this._scene.customRenderTargets.push(this._oldAccumulationRT);
+
+        // Now, create the accumulation pass
+        const ppOptions: PostProcessOptions = {
+            width: this._engine.getRenderWidth(),
+            height: this._engine.getRenderHeight(),
+            textureFormat: Constants.TEXTUREFORMAT_RGBA,
+            textureType: Constants.TEXTURETYPE_UNSIGNED_BYTE,
+            samplingMode: Constants.TEXTURE_NEAREST_SAMPLINGMODE,
+            uniforms: ["accumulationParameters"],
+            samplers: ["oldAccumulationSampler", "prevLocalPositionSampler", "localPositionSampler", "motionSampler"],
+            engine: this._engine,
+            reusable: false,
+        };
+        this._outputPP = new PostProcess("accumulationPassPP", "iblShadowAccumulation", ppOptions);
+        this._outputPP.autoClear = false;
+        this._outputPP.onApply = (effect) => {
+            this._updatePostProcess(effect);
+        };
+    }
+
+    public _updatePostProcess(effect: Effect) {
+        effect.setVector4("accumulationParameters", new Vector4(this.remenance, this.reset ? 1.0 : 0.0, 0.0, 0.0));
+        effect.setTexture("oldAccumulationSampler", this._oldAccumulationRT);
+        effect.setTexture("prevLocalPositionSampler", this._oldLocalPositionRT);
+        // effect.setTexture("shadowSampler", this._renderPipeline.getBlurShadowTexture()!);
+
+        const prePassRenderer = this._scene.prePassRenderer;
+        if (prePassRenderer) {
+            // prePassRenderer.getRenderTarget().skipInitialClear = true;
+            const localPositionIndex = prePassRenderer.getIndex(Constants.PREPASS_LOCAL_POSITION_TEXTURE_TYPE);
+            if (localPositionIndex >= 0) effect.setTexture("localPositionSampler", prePassRenderer.getRenderTarget().textures[localPositionIndex]);
+            const velocityIndex = prePassRenderer.getIndex(Constants.PREPASS_VELOCITY_TEXTURE_TYPE);
+            if (velocityIndex >= 0) effect.setTexture("motionSampler", prePassRenderer.getRenderTarget().textures[velocityIndex]);
+        }
+
+        this.reset = false;
+    }
+
+    /** Called by render pipeline when canvas resized. */
+    public resize() {
+        this._oldAccumulationRT.resize({ width: this._engine.getRenderWidth(), height: this._engine.getRenderHeight() });
+        this._oldLocalPositionRT.resize({ width: this._engine.getRenderWidth(), height: this._engine.getRenderHeight() });
+    }
+
+    private _disposeTextures() {
+        this._oldAccumulationRT.dispose();
+        this._oldLocalPositionRT.dispose();
+    }
+
+    /**
+     * Checks if the pass is ready
+     * @returns true if the pass is ready
+     */
+    public isReady() {
+        return this._outputPP.isReady();

```

Comment:
I can see only 1 PP being awaited for readyness but 3 have been created in the file. Same goes for dispose, is it intended?

Reply:
Thanks for pointing this out. I've fixed these now.